pkg/storage/etcd3/watcher.go

```diff
@@ -52,6 +52,7 @@ type watchChan struct {
 	initialRev        int64
 	recursive         bool
 	filter            storage.FilterFunc
+	filterAcceptsAll  bool
```

Comment:
Well - that would mean that whenever we are accessing wc.filter, we need to check whether it is nil or not. This will complicate the code in my opinion.